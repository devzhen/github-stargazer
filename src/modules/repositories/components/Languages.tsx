import * as R from 'ramda';
import React from 'react';
import styled from 'styled-components/native';

import Message from '@components/Message';
import Theme from '@theme/index';

interface LanguagesProps {
  languages: any[];
  isFetchingNow: boolean;
}

const Languages = ({ languages, isFetchingNow }: LanguagesProps) => (
  <Container>
    <Message
      alignment="left"
      message="Languages:"
      offsetBottom={30}
      offsetTop={30}
      size={17}
      color={Theme.colors.orange}
    />
    {isFetchingNow && <Loader size="small" color="#eb9e38" />}
    {!isFetchingNow && R.isEmpty(languages) && (
      <Message
        alignment="left"
        message="There are no languages yet."
        offsetBottom={10}
        color={Theme.colors.lightBlack}
      />
    )}
    {!isFetchingNow &&
      languages.map((item: any) => {
        const name = R.prop('name', item);
        const percent = R.prop('percent', item);

        const message = `${name}: ${percent}`;

        return (
          <Message
            alignment="left"
            key={message}
            message={message}
            offsetBottom={10}
            color={Theme.colors.lightBlack}
          />
        );
      })}
  </Container>
);

const Container = styled.View`
  flex: 1;
  align-items: flex-start;
`;

const Loader = styled.ActivityIndicator`
  margin-left: auto;
  margin-right: auto;
`;

export default Languages;
