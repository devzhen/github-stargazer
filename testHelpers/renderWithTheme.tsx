import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components/native';
import 'jest-styled-components';

import theme from '@theme/index';

const renderWithTheme = (component: any) =>
  renderer.create(<ThemeProvider theme={theme}>{component}</ThemeProvider>);

export default renderWithTheme;
