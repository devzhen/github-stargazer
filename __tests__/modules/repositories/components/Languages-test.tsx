import React from 'react';
import 'jest-styled-components';

import Languages from '@modules/repositories/components/Languages';

// @ts-ignore
import renderWithTheme from '@testHelpers/renderWithTheme';

const langs = [
  { name: 'Javascript', percent: '20%' },
  { name: 'F#', percent: '20%' },
  { name: 'C#', percent: '20%' },
  { name: 'C++', percent: '20%' },
];

describe('Languages repository component', () => {
  it('renders correctly, 1 case', () => {
    const tree = renderWithTheme(
      <Languages languages={langs} isFetchingNow={true} />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly, 2 case', () => {
    const tree = renderWithTheme(
      <Languages languages={langs} isFetchingNow={false} />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly, 3 case', () => {
    const tree = renderWithTheme(
      <Languages languages={[]} isFetchingNow={false} />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
