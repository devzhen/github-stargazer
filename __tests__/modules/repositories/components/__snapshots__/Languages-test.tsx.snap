// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Languages repository component renders correctly, 1 case 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "flex-start",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
>
  <View
    offsetBottom={30}
    offsetTop={30}
    style={
      Array [
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "marginBottom": 30,
          "marginTop": 30,
          "paddingBottom": 0,
          "paddingLeft": 0,
          "paddingRight": 0,
          "paddingTop": 0,
        },
      ]
    }
  >
    <Text
      alignment="left"
      color="#eb9e38"
      onPress={[Function]}
      size={17}
      style={
        Array [
          Object {
            "color": "#eb9e38",
            "fontSize": 17,
            "fontWeight": "bold",
            "textAlign": "left",
          },
        ]
      }
      weight="bold"
    >
      Languages:
    </Text>
  </View>
  <ActivityIndicator
    animating={true}
    color="#eb9e38"
    hidesWhenStopped={true}
    size="small"
    style={
      Array [
        Object {
          "marginLeft": "auto",
          "marginRight": "auto",
        },
      ]
    }
  />
</View>
`;

exports[`Languages repository component renders correctly, 2 case 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "flex-start",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
>
  <View
    offsetBottom={30}
    offsetTop={30}
    style={
      Array [
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "marginBottom": 30,
          "marginTop": 30,
          "paddingBottom": 0,
          "paddingLeft": 0,
          "paddingRight": 0,
          "paddingTop": 0,
        },
      ]
    }
  >
    <Text
      alignment="left"
      color="#eb9e38"
      onPress={[Function]}
      size={17}
      style={
        Array [
          Object {
            "color": "#eb9e38",
            "fontSize": 17,
            "fontWeight": "bold",
            "textAlign": "left",
          },
        ]
      }
      weight="bold"
    >
      Languages:
    </Text>
  </View>
  <View
    offsetBottom={10}
    offsetTop={0}
    style={
      Array [
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "marginBottom": 10,
          "marginTop": 0,
          "paddingBottom": 0,
          "paddingLeft": 0,
          "paddingRight": 0,
          "paddingTop": 0,
        },
      ]
    }
  >
    <Text
      alignment="left"
      color="#383838"
      onPress={[Function]}
      size={14}
      style={
        Array [
          Object {
            "color": "#383838",
            "fontSize": 14,
            "fontWeight": "bold",
            "textAlign": "left",
          },
        ]
      }
      weight="bold"
    >
      Javascript: 20%
    </Text>
  </View>
  <View
    offsetBottom={10}
    offsetTop={0}
    style={
      Array [
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "marginBottom": 10,
          "marginTop": 0,
          "paddingBottom": 0,
          "paddingLeft": 0,
          "paddingRight": 0,
          "paddingTop": 0,
        },
      ]
    }
  >
    <Text
      alignment="left"
      color="#383838"
      onPress={[Function]}
      size={14}
      style={
        Array [
          Object {
            "color": "#383838",
            "fontSize": 14,
            "fontWeight": "bold",
            "textAlign": "left",
          },
        ]
      }
      weight="bold"
    >
      F#: 20%
    </Text>
  </View>
  <View
    offsetBottom={10}
    offsetTop={0}
    style={
      Array [
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "marginBottom": 10,
          "marginTop": 0,
          "paddingBottom": 0,
          "paddingLeft": 0,
          "paddingRight": 0,
          "paddingTop": 0,
        },
      ]
    }
  >
    <Text
      alignment="left"
      color="#383838"
      onPress={[Function]}
      size={14}
      style={
        Array [
          Object {
            "color": "#383838",
            "fontSize": 14,
            "fontWeight": "bold",
            "textAlign": "left",
          },
        ]
      }
      weight="bold"
    >
      C#: 20%
    </Text>
  </View>
  <View
    offsetBottom={10}
    offsetTop={0}
    style={
      Array [
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "marginBottom": 10,
          "marginTop": 0,
          "paddingBottom": 0,
          "paddingLeft": 0,
          "paddingRight": 0,
          "paddingTop": 0,
        },
      ]
    }
  >
    <Text
      alignment="left"
      color="#383838"
      onPress={[Function]}
      size={14}
      style={
        Array [
          Object {
            "color": "#383838",
            "fontSize": 14,
            "fontWeight": "bold",
            "textAlign": "left",
          },
        ]
      }
      weight="bold"
    >
      C++: 20%
    </Text>
  </View>
</View>
`;

exports[`Languages repository component renders correctly, 3 case 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "flex-start",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
>
  <View
    offsetBottom={30}
    offsetTop={30}
    style={
      Array [
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "marginBottom": 30,
          "marginTop": 30,
          "paddingBottom": 0,
          "paddingLeft": 0,
          "paddingRight": 0,
          "paddingTop": 0,
        },
      ]
    }
  >
    <Text
      alignment="left"
      color="#eb9e38"
      onPress={[Function]}
      size={17}
      style={
        Array [
          Object {
            "color": "#eb9e38",
            "fontSize": 17,
            "fontWeight": "bold",
            "textAlign": "left",
          },
        ]
      }
      weight="bold"
    >
      Languages:
    </Text>
  </View>
  <View
    offsetBottom={10}
    offsetTop={0}
    style={
      Array [
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "marginBottom": 10,
          "marginTop": 0,
          "paddingBottom": 0,
          "paddingLeft": 0,
          "paddingRight": 0,
          "paddingTop": 0,
        },
      ]
    }
  >
    <Text
      alignment="left"
      color="#383838"
      onPress={[Function]}
      size={14}
      style={
        Array [
          Object {
            "color": "#383838",
            "fontSize": 14,
            "fontWeight": "bold",
            "textAlign": "left",
          },
        ]
      }
      weight="bold"
    >
      There are no languages yet.
    </Text>
  </View>
</View>
`;
